File: codewars/cpp/square_n_sum.cpp
// https://www.codewars.com/kata/515e271a311df0350d00000f/train/cpp


#include <iostream>
#include <vector>
#include <cmath>

int square_sum(const std::vector<int>& numbers) {
    int sum = 0;

    for(auto i : numbers) {
        sum += pow(i, 2);
    }

    return sum;
}

int main() {
    int limit, values;
    std::vector<int> numbers;

    std::cin >> limit;
    for(int i = 0; i < limit; ++i) {
        std::cin >> values;
        numbers.push_back(values);
    }

    std::cout << square_sum(numbers) << std::endl;

    return 0;
}

File: codewars/js/square_n_sum.js
// https://www.codewars.com/kata/515e271a311df0350d00000f/train/javascript

function squareSum(numbers) {
    let sum = 0;

    numbers.forEach(element => {
        sum += Math.pow(element, 2);
    });

    return sum;
}

console.log(squareSum([1, 2, 2]));
